sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/FieldBuilder.java

```diff
@@ -0,0 +1,259 @@
+        return SUPPORTED_NONE_PARAMETERIZED_TYPE.containsKey(type);
+    }
+
+    private static Field buildNoneParameterizedType(java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        Field searchField = convertToBasicSearchField(classField, logger);
+        return enrichWithAnnotation(searchField, classField, logger);
+    }
+
+
+    private static boolean isArrayOrList(Type type) {
+        if (type.getClass().isArray()) {
+            return true;
+        }
+        return isList(type);
+    }
+
+    private static boolean isList(Type type) {
+        if (type instanceof ParameterizedType) {
+            Type rawType = ((ParameterizedType) type).getRawType();
+            return List.class.isAssignableFrom((Class<?>) rawType);
+        }
+        return false;
+    }
+
+    private static Field buildCollectionField(java.lang.reflect.Field classField,
+        Stack<Class<?>> classChain, ClientLogger logger) {
+        Type componentOrElementType = getComponentOrElementType(classField.getGenericType(), logger);
+        if (isSupportedNoneParameterizedType(componentOrElementType)) {
+            Field searchField = convertToBasicSearchField(classField, logger);
+            return enrichWithAnnotation(searchField, classField, logger);
+        }
+        List<Field> childFields = build((Class<?>) componentOrElementType, classChain, logger);
+        Field searchField = convertToBasicSearchField(classField, logger);
+        searchField.setFields(childFields);
+        return searchField;
+    }
+
+    private static Type getComponentOrElementType(Type arrayOrListType, ClientLogger logger) {
+        if (arrayOrListType.getClass().isArray()) {
+            return arrayOrListType.getClass().getComponentType();
+        }
+        if (isList(arrayOrListType)) {
+            ParameterizedType pt = (ParameterizedType) arrayOrListType;
+            return pt.getActualTypeArguments()[0];
+        }
+        throw logger.logExceptionAsError(new RuntimeException("Should not be there"));
+    }
+
+    private static Field convertToBasicSearchField(java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        Field searchField = new Field();
+        searchField.setName(classField.getName());
+        DataType dataType = covertToDataType(classField.getGenericType(), false, logger);
+        searchField.setType(dataType)
+            .setKey(false)
+            .setSearchable(false)
+            .setFacetable(false)
+            .setHidden(false)
+            .setFilterable(false)
+            .setSortable(false);
+        return searchField;
+    }
+
+    private static Field enrichWithAnnotation(Field searchField, java.lang.reflect.Field classField,
+        ClientLogger logger) {
+        if (classField.isAnnotationPresent(SimpleFieldProperty.class)
+            && classField.isAnnotationPresent(SearchableFieldProperty.class)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
```

Comment:
Doesn't needed to be changed now, but does this make more sense as an `IllegalStateException`?

Reply:
This means user uses both annotation at the same time. This can be fixed by customer.

Reply:
`IllegalStateException` sounds like not able to fix by customer.